magtogoek/adcp/process.py

```diff
@@ -670,12 +670,24 @@ def _regrid_dataset(dataset: xr.Dataset, pconfig: ProcessConfig) -> xr.Dataset:
     and with equal weights for all data inside each bin.
  
     """
+    # Apply quality control
+    for var_ in 'uvw':
```

Comment:
I can see why setting values with flag > 2 to nan is the only way to go. However, since this replaces data, a note should be added in the documentation and a log message to warm/inform the user of it.

Reply:
I agree. Since there is an `interpolated_value` flag already, this would seem to be a use case for it.